src/object/cli_obj.c

```diff
@@ -6474,15 +6474,16 @@ dc_obj_query_key(tse_task_t *api_task)
 	for (i = grp_idx; i < grp_idx + grp_nr; i++) {
 		int start_shard;
 		int j;
+		int shard_cnt = 0;
 
 		/* Try leader for current group */
 		if (!obj_is_ec(obj) || (obj_is_ec(obj) && !obj->cob_ec_parity_rotate)) {
```

Comment:
Some redundant check for EC?
if (!obj_is_ec(obj) || !obj->cob_ec_parity_rotate)

Reply:
"if (!obj->cob_ec_parity_rotate)" seems enough.